Gems/LmbrCentral/Code/Tests/EditorBoxShapeComponentTests.cpp

```diff
@@ -56,5 +67,111 @@ namespace LmbrCentral
 
        EXPECT_EQ(dimensions, AZ::Vector3(0.37f, 0.57f, 0.66f));
     }
+
+    class EditorBoxShapeComponentFixture : public UnitTest::ToolsApplicationFixture
+    {
+    public:
+        void SetUpEditorFixtureImpl() override;
+        void TearDownEditorFixtureImpl() override;
+
+        AZStd::unique_ptr<AZ::ComponentDescriptor> m_editorBoxShapeComponentDescriptor;
+        AZStd::unique_ptr<AZ::ComponentDescriptor> m_editorSphereShapeComponentDescriptor;
+
+        AZ::Entity* m_entity = nullptr;
+    };
+
+    void EditorBoxShapeComponentFixture::SetUpEditorFixtureImpl()
+    {
+        AZ::SerializeContext* serializeContext = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(serializeContext, &AZ::ComponentApplicationBus::Events::GetSerializeContext);
+
+        // need to reflect EditorSphereShapeComponent in order for EditorBaseShapeComponent to be reflected
+        m_editorSphereShapeComponentDescriptor = AZStd::unique_ptr<AZ::ComponentDescriptor>(EditorSphereShapeComponent::CreateDescriptor());
+
+        m_editorBoxShapeComponentDescriptor =
+            AZStd::unique_ptr<AZ::ComponentDescriptor>(EditorBoxShapeComponent::CreateDescriptor());
+
+        ShapeComponentConfig::Reflect(serializeContext);
+        BoxShape::Reflect(serializeContext);
+        m_editorSphereShapeComponentDescriptor->Reflect(serializeContext);
+        m_editorBoxShapeComponentDescriptor->Reflect(serializeContext);
+
+        UnitTest::CreateDefaultEditorEntity("BoxShapeComponentEntity", &m_entity);
+        m_entity->Deactivate();
+        m_entity->CreateComponent(AzToolsFramework::Components::EditorNonUniformScaleComponent::RTTI_Type());
+        m_entity->CreateComponent(EditorBoxShapeComponentTypeId);
+        m_entity->Activate();
+    }
+
+    void EditorBoxShapeComponentFixture::TearDownEditorFixtureImpl()
+    {
+        AzToolsFramework::EditorEntityContextRequestBus::Broadcast(
+            &AzToolsFramework::EditorEntityContextRequestBus::Events::DestroyEditorEntity, m_entity->GetId());
+        m_entity = nullptr;
+
+        m_editorBoxShapeComponentDescriptor.reset();
+        m_editorSphereShapeComponentDescriptor.reset();
+    }
+
+    using EditorBoxShapeComponentManipulatorFixture =
+        UnitTest::IndirectCallManipulatorViewportInteractionFixtureMixin<EditorBoxShapeComponentFixture>;
+
+    TEST_F(EditorBoxShapeComponentManipulatorFixture, BoxShapeNonUniformScaleManipulatorsScaleCorrectly)
+    {
+        // a rotation which should map (1, 0, 0) to (0.8, 0.6, 0)
```

Comment:
question: Not sure I fully understand this comment sorry, could it be expanded here on the other values used below?

Reply:
It just means it rotates the x axis to the direction on the right. I tried to choose transformations which would give "nice" numbers to work with rather than ones with lots of decimal places.

Reply:
Okay cool, potentially adding that note as a comment would help future readers but it's not critical